data/systemd/Makefile

```diff
@@ -14,29 +14,48 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 SNAPD_ENVIRONMENT_FILE := /etc/environment
-SNAP_MOUNT_DIR := /snap
+SNAP_MOUNT_DIR ?= /snap
 BINDIR := /usr/bin
 LIBEXECDIR := /usr/lib
 SYSTEMDSYSTEMUNITDIR := /lib/systemd/system
 
-SYSTEMD_UNITS_GENERATED := $(patsubst %.service.in,%.service,$(wildcard *.service.in))
-SYSTEMD_UNITS := ${SYSTEMD_UNITS_GENERATED} $(wildcard *.timer) $(wildcard *.socket)
+SYSTEMD_UNITS_GENERATED := $(wildcard *.in)
+SYSTEMD_UNITS = $(SYSTEMD_UNITS_GENERATED:.in=) $(wildcard *.timer) $(wildcard *.socket)
 
-%.service: %.service.in
-	cat $< | \
-		sed s:@libexecdir@:${LIBEXECDIR}:g | \
-		sed s:@SNAPD_ENVIRONMENT_FILE@:${SNAPD_ENVIRONMENT_FILE}:g | \
-		sed s:@bindir@:${BINDIR}:g | \
-		sed s:@SNAP_MOUNT_DIR@:${SNAP_MOUNT_DIR}:g  | \
-		cat > $@
+# The special mount unit for "/snap" must be named after the path and the path
+# is a variable.  Compute it in make for simplicity. This is equivalent to
+# calling systemd-escape --path $(SNAP_MOUNT_DIR)
+snap_mount_unit=$(subst /,-,$(patsubst %/,%,$(patsubst /%,%,$(SNAP_MOUNT_DIR)))).mount
 
-all: ${SYSTEMD_UNITS}
+# Yes, we want the extra unit too, thank you
+SYSTEMD_UNITS += $(snap_mount_unit)
 
-install: $(SYSTEMD_UNITS)
+.PHONY: all
+all: $(SYSTEMD_UNITS)
+
+.PHONY: install
+install: $(filter-out snap.mount,$(SYSTEMD_UNITS)) $(snap_mount_unit)
 	# NOTE: old (e.g. 14.04) GNU coreutils doesn't -D with -t
-	install -d -m 0755 ${DESTDIR}/${SYSTEMDSYSTEMUNITDIR}
-	install -m 0644 -t ${DESTDIR}/${SYSTEMDSYSTEMUNITDIR} $^
-	install -m 0755 -t ${DESTDIR}/${LIBEXECDIR}/snapd snapd.core-fixup.sh
+	install -d -m 0755 $(DESTDIR)/$(SYSTEMDSYSTEMUNITDIR)
+	install -m 0644 -t $(DESTDIR)/$(SYSTEMDSYSTEMUNITDIR) $^
+	install -d -m 0755 $(DESTDIR)/$(LIBEXECDIR)/snapd
+	install -m 0755 -t $(DESTDIR)/$(LIBEXECDIR)/snapd snapd.core-fixup.sh
 
+.PHONY: clean
 clean:
-	rm -f ${SYSTEMD_UNITS_GENERATED}
+	rm -f $(SYSTEMD_UNITS_GENERATED:.in=) $(snap_mount_unit)
+
+%: %.in
+	cat $< | \
+		sed s:@libexecdir@:$(LIBEXECDIR):g | \
+		sed s:@SNAPD_ENVIRONMENT_FILE@:$(SNAPD_ENVIRONMENT_FILE):g | \
+		sed s:@bindir@:$(BINDIR):g | \
+		sed s:@SNAP_MOUNT_DIR@:$(SNAP_MOUNT_DIR):g  | \
+		cat > $@
+
+# If SNAP_MOUNT_DIR uses non-default location then just copy the file.
```

Comment:
Maybe: `If SNAP_MOUNT_DIR uses non-default location then use snap.mount as the template to generate the matching mount unit`?

Reply:
It's not much generate as it's already generated but I'll tweak your suggestion and use it.